core/amber/src/main/python/core/proxy/proxy_server.py

```diff
@@ -152,8 +153,14 @@ def do_put(
 
         data: Table = reader.read_all()
         command: bytes = descriptor.command
-        logger.debug(f"getting a data batch {data}")
-        self.process_data(command, data)
+        logger.info(f"getting a data batch {data}")
+
+        num_batches_in_queue = self.process_data(command, data)
+        if isinstance(num_batches_in_queue, int):
+            num_batches_buf: Buffer = py_buffer(
+                num_batches_in_queue.to_bytes(length=8, byteorder="little")
+            )
```

Comment:
what are the other possibilities? i.e., what should we do for else case, or would it ever happen?

Reply:
@Yicong-Huang `num_batches_in_queue` should always be an integer because process_data should always return back an integer credit value.  There should be no other possibilities if self.process_data correctly returns the credit integer value.  We only added this check because in some test cases, we don't get an integer and it causes an error to occur when we try converting it to bytes.

Should I add some assertions or error logging in the cases it is not an instance (even though it shouldn't be possible) ?